### PR TITLE
JMock SMS and Service tasks

### PR DESCRIPTION
**Pull request by kristiyanpetkov**

### CODE REVIEW

File: src/com/clouway/Sms/Reciever.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.clouway.Sms;
+
+/**
+ * Created by clouway on 15-9-25.
+ */
+public class Reciever {
```
Comment (mgenov) [inline src/com/clouway/Sms/Reciever.java]: Recipient sounds better in the context of SMS messaging.

File: src/com/clouway/Sms/Gateway.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.clouway.Sms;
+
+/**
+ * Created by clouway on 15-9-25.
+ */
+public interface Gateway {
+    void recieveMessage(Message message, Reciever reciever);
```
Comment (mgenov) [inline src/com/clouway/Sms/Gateway.java]: Probably App should send sms messages to the SmsGateway and it should send them to mobile devices.

File: test/com/clouway/Sms/MessageSenderTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        final Reciever reciever = new Reciever("0883497259");
+        final Sender sender = new Sender(reciever, gateway);
+        context.checking(new Expectations() {{
+            never(gateway).recieveMessage(message, reciever);
+        }});
+        sender.sendMessage(message);
+    }
+
+    @Test
+    public void TryToSendMessageWithEmptyHeader() {
```
Comment (mgenov) [inline test/com/clouway/Sms/MessageSenderTest.java]: method names are with lower case. 

https://github.com/clouway/teamspace/wiki/Java%20Code%20Convention

File: src/com/clouway/Sms/Reciever.java (hunk 1 above)
Comment (mgenov) [inline src/com/clouway/Sms/Reciever.java]: This is what google translate shows:

screen shot 2015-09-28 at 11 23 14 am

**Comment by mgenov** [comment]
As another PR, I would like to see in separate package, making of same test using FakeGateway, so we may compare the results.

File: src/com/clouway/Sms/Reciever.java (hunk 1 above)
Comment (kristiyanpetkov) [inline src/com/clouway/Sms/Reciever.java]: Yes, it sounds better. Fixed!

File: src/com/clouway/Sms/Gateway.java (hunk 2 above)
Comment (kristiyanpetkov) [inline src/com/clouway/Sms/Gateway.java]: Yes, it doesn't make sense. Method name changed to "sendMessage".

File: test/com/clouway/Sms/MessageSenderTest.java (hunk 3 above)
Comment (kristiyanpetkov) [inline test/com/clouway/Sms/MessageSenderTest.java]: Sorry for that newbie mistake! Fixed!

**Comment by mgenov** [comment]
Are you ready with your change ?

**Comment by mgenov** [comment]
https://help.github.com/articles/about-notifications/

**Comment by kristiyanpetkov** [comment]
@mgenov , i am ready with my changes.

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * Created by clouway on 15-9-29.
+ */
+public class FakeMessageSenderTest {
+    @Test
+    public void happyPath(){
+        FakeSmsGateway fakeSmsGateway = new FakeSmsGateway();
+        Message message = new Message("Zdravey","Kak si?");
+        Recipient recipient = new Recipient("0883497259");
+        fakeSmsGateway.sendMessage(message,recipient);
+        assertThat("Zdravey", is(equalTo(fakeSmsGateway.headerText)));
```
Comment (mgenov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: Doesn't look like a good design, but you are on right direction. 

assertThat(fakeSmsGateway.lastSendMessage().headerText, is(equalTo("Zdravey"));

Why you have only one test ?

Comment (kristiyanpetkov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: I just added all tests.

**Comment by kristiyanpetkov** [comment]
@mgenov , i am ready with the second task from Testing with Mocks lesson.

File: src/com/clouway/Service/DataBase.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.clouway.Service;
+
+/**
+ * Created by clouway on 15-9-30.
+ */
+public interface DataBase {
```
Comment (mgenov) [inline src/com/clouway/Service/DataBase.java]: Name is common, where you are dealing with concrete type Person. This is wrong and need to be fixed to concrete or to common. 

We already discussed this several times, so I'll wait for your suggestion.

Comment (kristiyanpetkov) [inline src/com/clouway/Service/DataBase.java]: DataBaseForPeopleOver10andUnder100YearsOld. Does it sounds better?

Comment (mgenov) [inline src/com/clouway/Service/DataBase.java]: Are you kidding with me?

Comment (kristiyanpetkov) [inline src/com/clouway/Service/DataBase.java]: I just shoot in the dark.  
Are PersonDataBase or PeopleDataBase more concrete?

Comment (mgenov) [inline src/com/clouway/Service/DataBase.java]: PersonRepository, PersonDatabase, PersonStore. 

Have you read something about this in GOOS Book? GOOS Book has chapter about this stuff.

Comment (kristiyanpetkov) [inline src/com/clouway/Service/DataBase.java]: I just flipped through the pages. I will take a deeper look at it.
Interface name already changed to PersonDatabase.

File: src/com/clouway/Service/Service.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package com.clouway.Service;
+
+/**
+ * Created by clouway on 15-9-30.
+ */
+public class Service {
```
Comment (mgenov) [inline src/com/clouway/Service/Service.java]: Service is common, where PersonDatabase is concrete, so naming should be changed. 

If I wrote this, I would remove this class and use only PersonDatabase and Validator to perform registrations.

File: src/com/clouway/Service/Service.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    private final PersonDatabase personDatabase;
+    private final Validator validator;
+
+    public Service(Validator validator, PersonDatabase personDatabase) {
+        this.validator = validator;
+        this.personDatabase = personDatabase;
+    }
+
+    public void checkIfPersonIsSuitableToBeAddedInDataBase(Person person) {
```
Comment (mgenov) [inline src/com/clouway/Service/Service.java]: register(Person person) sounds more reasonable. As you can see in the body of this method, you are doing and registration, but the method says that it just checks something.

File: src/com/clouway/Service/Service.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        if (validator.validateIfAgeIsOver10AndUnder100(person.age)) {
+            personDatabase.addPersonToDataBase(person);
+        }
+    }
+
+    public boolean checkIfAgeIsUnderOrOver18Years(Person person) {
```
Comment (mgenov) [inline src/com/clouway/Service/Service.java]: isAdult(Person person( sounds more reasonable.

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        Recipient recipient = new Recipient("0883497259");
+        Sender sender = new Sender(recipient,fakeSmsGateway);
+        sender.sendMessage(message);
+        assertThat("Zdravey", is(equalTo(fakeSmsGateway.headerText)));
+        assertThat("Kak si?",is(equalTo(fakeSmsGateway.messageText)));
+        assertThat("0883497259",is(equalTo(fakeSmsGateway.recipientNumber)));
+    }
+
+    @Test
+    public void tryToSendMessageWithMoreThanOneHundredAndTwentySymbols(){
```
Comment (mgenov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: name could be changed to something simpler like: sendLongMessage

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+        Recipient recipient = new Recipient("0883497259");
+        Sender sender = new Sender(recipient,fakeSmsGateway);
+        sender.sendMessage(message);
+        assertThat(null, is(equalTo(fakeSmsGateway.headerText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.messageText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.recipientNumber)));
+    }
+
+    @Test
+    public void tryToSendMessageWithLessThanOneSymbol(){
```
Comment (mgenov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: name could be changed to shortened one: sendShortMessage or messageIsTooShort, i.e checking the scenario when message is too short.

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        Recipient recipient = new Recipient("0883497259");
+        Sender sender = new Sender(recipient,fakeSmsGateway);
+        sender.sendMessage(message);
+        assertThat(null, is(equalTo(fakeSmsGateway.headerText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.messageText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.recipientNumber)));
+    }
+
+    @Test
+    public void tryToSendMessageWithEmptyHeader(){
```
Comment (mgenov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: emptyHeader

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+        Recipient recipient = new Recipient("0883497259");
+        Sender sender = new Sender(recipient,fakeSmsGateway);
+        sender.sendMessage(message);
+        assertThat(null, is(equalTo(fakeSmsGateway.headerText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.messageText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.recipientNumber)));
+    }
+
+    @Test
+    public void tryToSendMessageWithEmptyReciever(){
```
Comment (mgenov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: emptyReceiver

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+        Sender sender = new Sender(recipient,fakeSmsGateway);
+        sender.sendMessage(message);
+        assertThat(null, is(equalTo(fakeSmsGateway.headerText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.messageText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.recipientNumber)));
+    }
+
+    @Test
+    public void tryToSendMessageWithNullText(){
+        Message message = new Message("Zdravey",null);
```
Comment (mgenov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: textIsNull

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+        Recipient recipient = new Recipient("");
+        Sender sender = new Sender(recipient,fakeSmsGateway);
+        sender.sendMessage(message);
+        assertThat(null, is(equalTo(fakeSmsGateway.headerText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.messageText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.recipientNumber)));
+    }
+
+    @Test
+    public void tryToSendMessageWithNullHeader(){
```
Comment (mgenov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: headerIsNull

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+        Recipient recipient = new Recipient("");
+        Sender sender = new Sender(recipient,fakeSmsGateway);
+        sender.sendMessage(message);
+        assertThat(null, is(equalTo(fakeSmsGateway.headerText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.messageText)));
+        assertThat(null,is(equalTo(fakeSmsGateway.recipientNumber)));
+    }
+
+    @Test
+    public void tryToSendMessageWithNullRecipient(){
```
Comment (mgenov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: recipientIsNull

File: src/com/clouway/Sms/Sender.java [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+    public Sender(Recipient recipient, SmsGateway smsGateway) {
+        this.reciever = recipient;
+        this.smsGateway = smsGateway;
+    }
+
+    public void sendMessage(Message message) {
+        if (message.text == null || message.header == null || reciever.getPhoneNumber() == null) {
+            return;
+        }
+        if (message.text.length() >= 1 && message.text.length() <= 120 && message.header.length() > 1 && reciever.getPhoneNumber().length() > 0) {
```
Comment (mgenov) [inline src/com/clouway/Sms/Sender.java]: message.text.length() is breaking the law of demeter. 

Try to refactore this by extracting some of this logic into the Message domain object.

File: src/com/clouway/Service/Validator.java [hunk 17]
```diff
@@ -0,0 +1,11 @@
+package com.clouway.Service;
+
+/**
+ * Created by clouway on 15-9-30.
+ */
+public interface Validator {
+    boolean validateIfAgeIsOver10AndUnder100(String age);
```
Comment (mgenov) [inline src/com/clouway/Service/Validator.java]: AgeValidator seems more reasonable name of the interface, cause Validator is too common, whether method names are concrete. 

Looks like a really bad design, cause if I have to add another boundary, I have to change the existing code and add another method following this pattern.

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java (hunk 15 above)
Comment (mgenov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: You can follow this pattern to fix all test names.

File: test/com/clouway/Service/ServiceTest.java [hunk 18]
```diff
@@ -0,0 +1,159 @@
...
+            oneOf(personDatabase).getByName(person.name);
+            will(returnValue(person));
+            oneOf(validator).validateIfAgeIsOver18(person.age);
+            will(returnValue(false));
+        }});
+        service.checkIfAgeIsUnderOrOver18Years(person);
+    }
+
+    @Test
+    public void testWhatHappensIfPersonIsNull() {
```
Comment (mgenov) [inline test/com/clouway/Service/ServiceTest.java]: "test" is name of the method was used in junit3 as we already discussed.

File: src/com/clouway/Service/Service.java (hunk 6 above)
Comment (kristiyanpetkov) [inline src/com/clouway/Service/Service.java]: Fixed!

File: src/com/clouway/Service/Service.java (hunk 7 above)
Comment (kristiyanpetkov) [inline src/com/clouway/Service/Service.java]: Fixed!

File: src/com/clouway/Service/Service.java (hunk 8 above)
Comment (kristiyanpetkov) [inline src/com/clouway/Service/Service.java]: Fixed!

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java (hunk 9 above)
Comment (kristiyanpetkov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: Fixed!

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java (hunk 10 above)
Comment (kristiyanpetkov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: Fixed!

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java (hunk 11 above)
Comment (kristiyanpetkov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: Fixed!

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java (hunk 12 above)
Comment (kristiyanpetkov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: Fixed!

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java (hunk 13 above)
Comment (kristiyanpetkov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: Fixed!

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java (hunk 14 above)
Comment (kristiyanpetkov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: Fixed!

File: test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java (hunk 15 above)
Comment (kristiyanpetkov) [inline test/com/clouway/FakeSmsGateway/FakeMessageSenderTest.java]: Fixed!

File: src/com/clouway/Sms/Sender.java (hunk 16 above)
Comment (kristiyanpetkov) [inline src/com/clouway/Sms/Sender.java]: Fixed!

File: src/com/clouway/Service/Validator.java (hunk 17 above)
Comment (kristiyanpetkov) [inline src/com/clouway/Service/Validator.java]: Fixed!

File: test/com/clouway/Service/ServiceTest.java (hunk 18 above)
Comment (kristiyanpetkov) [inline test/com/clouway/Service/ServiceTest.java]: Fixed!

File: src/com/clouway/Service/AgeValidator.java [hunk 19]
```diff
@@ -0,0 +1,11 @@
+package com.clouway.Service;
+
+/**
+ * Created by clouway on 15-9-30.
+ */
+public interface AgeValidator {
+    boolean validateIfAgeIsOver10AndUnder100(String age);
```
Comment (mgenov) [inline src/com/clouway/Service/AgeValidator.java]: Looks like this methods are still as they are.

Comment (kristiyanpetkov) [inline src/com/clouway/Service/AgeValidator.java]: Methods names changed to isValid(String age) and isAdult(String age)

File: test/com/clouway/Sms/MessageSenderTest.java [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+        final Sender sender = new Sender(recipient, smsGateway);
+        context.checking(new Expectations() {{
+            oneOf(smsGateway).sendMessage(message, recipient);
+        }});
+        sender.sendMessage(message);
+
+    }
+
+    @Test
+    public void tryToSendMessageWithMoreThanOneHundredAndTwentySymbols() {
```
Comment (mgenov) [inline test/com/clouway/Sms/MessageSenderTest.java]: it's long message, no matter when it's 100 or 1000 symbols long.

File: test/com/clouway/Sms/MessageSenderTest.java [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+            never(smsGateway).sendMessage(message, recipient);
+        }});
+        sender.sendMessage(message);
+    }
+
+    ;
+
+
+    @Test
+    public void tryToSendMessageWithLessThanOneSymbol() {
```
Comment (mgenov) [inline test/com/clouway/Sms/MessageSenderTest.java]: Test naming is still not fixed.

File: test/com/clouway/Service/PeopleRegisterServiceTest.java [hunk 22]
```diff
@@ -0,0 +1,159 @@
...
+        mockery.checking(new Expectations() {{
+            oneOf(ageValidator).isBetween10and100(person.age);
+            will(returnValue(true));
+            oneOf(personDatabase).addPersonToDataBase(person);
+        }});
+        peopleRegisterService.register(person);
+    }
+
+    @Test
+    public void ageUnder10() {
```
Comment (mgenov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: Tests here that are using these validation methods need to be fixed too.

File: test/com/clouway/Sms/MessageSenderTest.java (hunk 20 above)
Comment (kristiyanpetkov) [inline test/com/clouway/Sms/MessageSenderTest.java]: Method name changed to sendLongMessage().

File: test/com/clouway/Sms/MessageSenderTest.java (hunk 21 above)
Comment (kristiyanpetkov) [inline test/com/clouway/Sms/MessageSenderTest.java]: Fixed!

File: test/com/clouway/Service/PeopleRegisterServiceTest.java (hunk 22 above)
Comment (kristiyanpetkov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: Test method names or something else? I'm not sure i understand.

Comment (mgenov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: When you check that some value is between 2 other values, then you should have some of the following scenarios:

1. value is lower then lower limit  (out of boundaries)
2. value is higher then lower limit and lower then upper limit (in boundaries)
3. value is higher then high limit (out of boundaries) 
4. value is equal to low limit (in boundaries)
5. value is equal to high limit (in boundaries)

Comment (mgenov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: As you can see, the following scenarios are describing behaviour, not state.

Comment (kristiyanpetkov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: I added the missing tests and changed method names.

File: test/com/clouway/Service/PeopleRegisterServiceTest.java [hunk 23]
```diff
@@ -0,0 +1,184 @@
...
+        mockery.checking(new Expectations() {{
+            oneOf(ageValidator).isBetween10and100(person.age);
+            will(returnValue(true));
+            oneOf(personDatabase).addPersonToDataBase(person);
+        }});
+        peopleRegisterService.register(person);
+    }
+
+    @Test
+    public void ageLowerThanLowerLimit() {
```
Comment (mgenov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: You changed the name of the tests, but the method signature is still the same.

File: test/com/clouway/Service/PeopleRegisterServiceTest.java [hunk 24]
```diff
@@ -0,0 +1,184 @@
...
+        final Person person = new Person("Christiano Ronaldo Jr.", "9");
+        final PeopleRegisterService peopleRegisterService = new PeopleRegisterService(ageValidator, personDatabase);
+        mockery.checking(new Expectations() {{
+            oneOf(ageValidator).isBetween10and100(person.age);
```
Comment (mgenov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: What if I want persons 4 types of persons: 
1. from 20 to 60
2. from 10 to 18
3. from 30 to 50 
4. from 25 to 34

Do I have to make this combinations as methods in AgeValidator ? You can try to make it, to see what will happen in your test.

Comment (kristiyanpetkov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: i changed the validator method name to: isBetweenLowerAndHigherLimit() , so you dont have to make separate validators for all cases. You can use isBetweenLowerAndHigherLimit() method for all the cases.

File: test/com/clouway/Service/PeopleRegisterServiceTest.java (hunk 23 above)
Comment (kristiyanpetkov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: method name changed to:  isBetweenLowerAndHigherLimit()

File: test/com/clouway/Service/PeopleRegisterServiceTest.java (hunk 24 above)
Comment (kristiyanpetkov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: If you want, i can change the validator method name to: isBetween(String age, int lowerAge, int higherAge);

File: test/com/clouway/Service/PeopleRegisterServiceTest.java (hunk 23 above)
Comment (mgenov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: When you read this code that way, is it sounds good to you ?

Comment (kristiyanpetkov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: Not really. I think the proper name should be:
oneOf(ageValidator).validateIfAgeIsBetween10and100(person.age);

Comment (kristiyanpetkov) [inline test/com/clouway/Service/PeopleRegisterServiceTest.java]: or 
oneOf(ageValidator).validateIfAgeIsBetweenLowerAndHigherLimit(person.age);

**Comment by kristiyanpetkov** [comment]
I added BoundaryAgeValidator Test class.

File: test/com/clouway/service/BoundaryAgeValidatorTest.java [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Test
+    public void lowerThanLowerBoundary(){
+        final Person person = new Person("Unknown","6");
+        final BoundaryAgeValidator bulgarianAgeValidator = new BoundaryAgeValidator(10,80);
+        assertEquals(false,bulgarianAgeValidator.isAdult(person.age));
+    }
+
+    @Test
+    public void higherThanHigherBoundary(){
```
Comment (mgenov) [inline test/com/clouway/service/BoundaryAgeValidatorTest.java]: What if value is equal to the lower or upper boundary?

File: src/com/clouway/service/Person.java [hunk 26]
```diff
@@ -0,0 +1,14 @@
+package com.clouway.service;
+
+/**
+ * Created by clouway on 15-9-30.
+ */
+public class Person {
+    public String name;
```
Comment (mgenov) [inline src/com/clouway/service/Person.java]: public final

File: src/com/clouway/service/Person.java [hunk 27]
```diff
@@ -0,0 +1,14 @@
+package com.clouway.service;
+
+/**
+ * Created by clouway on 15-9-30.
+ */
+public class Person {
```
Comment (mgenov) [inline src/com/clouway/service/Person.java]: public final class Person, will protect for extending of this class.

File: test/com/clouway/service/BoundaryAgeValidatorTest.java (hunk 25 above)
Comment (kristiyanpetkov) [inline test/com/clouway/service/BoundaryAgeValidatorTest.java]: Age validation check will pass. Tests added.

File: src/com/clouway/service/Person.java (hunk 26 above)
Comment (kristiyanpetkov) [inline src/com/clouway/service/Person.java]: Fixed!

File: src/com/clouway/service/Person.java (hunk 27 above)
Comment (kristiyanpetkov) [inline src/com/clouway/service/Person.java]: Fixed!

**Comment by mgenov** [comment]
I cannot see fixes of public final stuff, cause you hadn't committed them. Only comment was added. 

LGTM

**Pull request merged**